go/internal/otel/otel.go

```diff
@@ -219,6 +218,10 @@ func createTracerProvider(
 		),
 		sdktrace.WithResource(resources),
 	}, opts...)
+	// Only configure a sampler when there is a sampling configuration.
+	if sampler != nil {
```

Comment:
This is different than the highlight SDK that always had a sampler with defaults.